jaeger-context/src/main/java/com/uber/jaeger/context/ThreadLocalTraceContext.java

```diff
@@ -27,7 +27,7 @@
 import java.util.Stack;
 
 public class ThreadLocalTraceContext implements TraceContext {
-  private final ThreadLocal<Stack<Span>> threadLocal = new SpanStackThreadLocal();
+  private static final ThreadLocal<Stack<Span>> threadLocal = new SpanStackThreadLocal();
```

Comment:
this is a change in semantics - there's no requirement that all instances of this class share the same thread-local

Reply:
You are absolutely right, I am removing this. Basically you bind thread locals per instance.